compress/src/main/c/Makefile

```diff
@@ -56,22 +56,27 @@ OBJS = \
 $(LIBNAME): $(OBJS)
 	$(CC) $(LD_FLAGS) -o $@ $(OBJS)
 
-build: $(LIBNAME) $(if $(OS:linux=),,$(if $(ARCH:x86_64=),,build32))
-
-build32:
-	$(MAKE) ARCH=i386 build
+buildAARCH64OnX86:
+	$(MAKE) CC=aarch64-linux-gnu-gcc ARCH=aarch64 build
+buildX86OnAARCH64:
+	$(MAKE) CC=x86_64-linux-gnu-gcc ARCH=x86_64 build
+build: $(LIBNAME) $(if $(OS:linux=),,$(if $(ARCH:$(REALARCH)=),,$(if $(ARCH:x86_64=),buildX86OnAARCH64,buildAARCH64OnX86)))
 
 $(DEST_DIR):
 	mkdir -p $(DEST_DIR)
 
-install: build $(DEST_DIR) $(if $(OS:linux=),,$(if $(ARCH:x86_64=),,install32))
+install: build $(DEST_DIR)  $(if $(OS:linux=),,$(if $(ARCH:$(REALARCH)=),,$(if $(ARCH:x86_64=),installX86,installAARCH64)))
 	cp $(LIBNAME) $(DEST_DIR)/$(subst -$(ARCH),,$(LIBNAME))
 
-install32:
-	$(MAKE) ARCH=i386 install
+installX86: 
```

Comment:
Why a separate target not just a variable?

Reply:
I opted for variables because it was cleaner and there wasn't a lot of value of allowing individual platform targets.